java-manta-client/src/main/java/com/joyent/manta/client/crypto/AesCtrCipherDetails.java

```diff
@@ -64,60 +64,57 @@ public boolean plaintextSizeCalculationIsAnEstimate() {
-                    startInclusive);
+            String msg = "Start position must be zero or higher (startInclusive=" + startInclusive + ")";
             throw new IllegalArgumentException(msg);
         }
 
-        if (startInclusive > plaintextMax) {
-            String msg = String.format("Start position must be less than maximum "
-                            + "ciphertext size [%d]. Actually: %d",
-                    plaintextMax, startInclusive);
+        if (startInclusive >= plaintextMax) {
+            String msg = "Start position must be less than maximum plaintext size (startInclusive="
+                    + startInclusive + ", plaintextMax=" + plaintextMax + ")";
             throw new IllegalArgumentException(msg);
         }
 
         if (endInclusive < 0) {
-            String msg = String.format("End position must be zero or higher. Actually: %d",
-                    endInclusive);
+            String msg = "End position must be zero or higher (endInclusive=" + endInclusive + ")";
             throw new IllegalArgumentException(msg);
         }
 
-        if (endInclusive > plaintextMax) {
-            String msg = String.format("End position must be less than maximum "
-                            + "ciphertext size [%d]. Actually: %d",
-                    plaintextMax, endInclusive);
+        if (endInclusive >= plaintextMax) {
+            String msg = "End position must be less than maximum plaintext size (endInclusive=" + endInclusive
+                    + ", plaintextMax=" + plaintextMax + ")";
             throw new IllegalArgumentException(msg);
         }
 
         final int blockSize = getBlockSizeInBytes();
-        long adjustedStart;
-        long adjustedEnd;
-        long plaintextEndLength;
 
-        // How many bytes do we offset in the first block
-        long plaintextStartAdjustment = startInclusive % blockSize;
+        // How many bytes do we offset in the first block?
+        final long plaintextBytesToSkipInitially = startInclusive % blockSize;
 
-        // The first block is considered block 0
-        double startBlockNumber = Math.floor(startInclusive / blockSize);
+        // Get the ciphertext block index for the start byte (zero based)
+        final long startingBlockNumberInclusive = startInclusive / blockSize;
 
-        // Content-Range: bytes=adjustedStart-adjustedEnd
-        adjustedStart = ((long) startBlockNumber) * blockSize;
+        // Get the ciphertext byte position for the start block (block size bytes per block)
+        final long ciphertextStartPositionInclusive = startingBlockNumberInclusive * blockSize;
 
-        // Get the full block for the end range, plus an additional block
-        double endBlockNumber = Math.ceil(endInclusive / blockSize) + 1;
+        // Get the ciphertext block index for the end byte (zero based)
+        final long endingBlockNumberInclusive = endInclusive / blockSize;
```

Comment:
With AES-CTR, ciphertext bytes directly line up with plaintext bytes.  Another way to describe is that the number of blocks is *always*...

    (end_block - start_block) + 1

AES-CTR isn't padded, so the last block in the range may not be an entire block if the range spans to the last block.  So you can't know the exact number of ciphertext bytes without knowing the size of the object.  But, per Range Request RFC (7233), requesting a larger range than exists is *not* an error as long as it begins in some existing range (only the existing range will be returned).

So, the number of bytes to request from manta in an AES-CTR range *can* always be...

    blocks * blocksize

or

    ((end_block - start_block) + 1) * blocksize

And, ciphertext start and end are...

     start_block * blocksize
     ((end_block + 1) * blocksize) - 1

And, length of plaintext with skip bytes is...

    request_end - (start_block * blocksize)

If you use these equations, you get the results I updated in `AesCtrCipherDetailsTest#translateByteRangeReturnsCorrectRange`.

The equations might not generalize to all ciphers, but AES-CTR is the only cipher effectively using it since `AesCtrCipherDetails` is the only implementation with the specific calculation (it's not implemented in `AbstractAesCipherDetails`).  Obviously these exclude any MAC.

These errors weren't off by one, the calculations were just incorrect.  Off by one had to do with comparing a zero based index to a byte count for maximum plaintext size (`AesCtrCipherDetails#L78`, `AesCtrCipherDetails#L89`).